sockeye/decoder.py

```diff
@@ -625,6 +638,16 @@ def decode_step(self,
         """
         source_encoded, prev_dynamic_source, source_encoded_length, prev_hidden, *layer_states = states
 
+        # Get last state from source (batch_size, num_target_embed)
+        enc_last_hidden = None
+        if self.config.enc_last_hidden_concat_to_embedding:
+            source_encoded_time_major = mx.sym.swapaxes(source_encoded,
```

Comment:
I think SequenceLast now supports an axis argument and you can avoid the explicit swapaxes. Could you check whether this can be simplified?

Reply:
I did not know that. Will do it.